### PR TITLE
Namespaces in effects

### PR DESCRIPTION
**Pull request by bugarela**
Hello :octocat: 

This fixes https://github.com/informalsystems/quint/issues/1288 by adding namespaces to state variables occurring inside effects. We use the same namespaces that are used for flattening, which guarantee that variables from different instances have different names. This enables the effect checker to distinguish between those variables which would otherwise be treated as the same thing.

I also hit a very weird bug with `run`, where flattening was breaking, although `test` and `verify` worked fine. Turns out that adding extra defs was resulting in something like `val myDef = fooval lastTrace = ...`, so the parser was parsing that up to `fooval` and ignoring the rest, but `fooval` couldn't be resolved while flattening. To solve this problem, I'm introducing a line break before the extra defs. (related to https://github.com/informalsystems/quint/issues/1052)

- [X] Tests added for any new code
- [X] Documentation added for any new functionality
- [x] Entries added to the respective `CHANGELOG.md` for any new functionality
- [ ] Feature table on [`README.md`](../README.md#roadmap) updated for any listed functionality

### CODE REVIEW

**Comment by shonfeder** [comment]
Quick question on the approach, given that

> We use the same namespaces that are used for flattening, which guarantee that variables from different instances have different names

Is there a fundamental reason that we cannot do the effect checking after we have run the flattening? If it were possible, the benefit would be reusing the existing namespacing logic, and having a more uniform data pipeline.

**Comment by shonfeder** [comment]
All this makes sense and is a good reason not to flatten first. The problem art that I don't like is that we are recomputing the same stringy encoding of name-spacing twice. Imo, it seems to me like we should only do this once, but we can leave that for later refactoring.

**Review by shonfeder — APPROVED** [review]
Looks great! Thanks also for the sync chat :D

File: quint/test/effects/namespaces.test.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { describe, it } from 'mocha'
```
[the remark this answers is not in the document]
Comment (bugarela) [reply quint/test/effects/namespaces.test.ts:1]: Yes! It was a bit cumbersome, but I did it

**Pull request merged**